bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,337 @@
+                             delimiter and before the ``close`` delimiter
+    '''
+    diff = Diff(file)
+
+    content_before = line[:match_object.start("open")]
+    content_after = line[match_object.end("close"):]
+
+    spacing = " " * required_spacing
+    replacement = (
+        "{before}{open}{spacing}{content}{spacing}{close}{after}".format(
+            before=content_before,
+            spacing=spacing,
+            after=content_after,
+            content=match_object.group("content").strip(),
+            open=match_object.group("open"),
+            close=match_object.group("close")))
+    diff.change_line(
+        line_number,
+        line,
+        replacement)
+    return {filename: diff}
+
+
+def generate_label_diff(file, filename, line, line_number,
+                        match_object, expected_label):
+    '''
+    Generates a diff for a missing or wrong control loop end label.
+    Missing labels will be added, wrong ones replaced, content after the
+    label will be left untouched.
+
+
+    :param match_object:   A Match object containing the groups ``close``
+                           containing the closing delimiters of a control
+                           end tag and the optional group ``label``
+                           containing the end tag label.
+    :param expected_label: The expected label for that control block.
+
+    '''
+    diff = Diff(file)
+
+    content_before = line[:match_object.end("close")]
+
+    suffix_start = match_object.end("close")
+    # if a label is there, we cut it out, by starting the kept suffix after it
+    if match_object.group("label") is not None:
+        suffix_start = match_object.end("label")
+
+    content_after = line[suffix_start:]
+
+    replacement = "{before}{label}{after}".format(
+        before=content_before,
+        after=content_after,
+        label=expected_label
+    )
+    diff.change_line(
+        line_number,
+        line,
+        replacement)
+    return {filename: diff}
+
+
+def is_spacing_valid(string, required_spacing):
+    '''
+    Checks if a string is properly spaced.
+
+    :param string:           The string which is to be checked.
+    :param required_spacing: The number of spaces expected on both
+                             sides of the string.
+    :return:                 True if the string has the required
+                             number of spaces on both sides, False
+                             otherwise.
+    '''
+    leading_spaces = len(string) - len(string.lstrip(' '))
+    trailing_spaces = len(string) - len(string.rstrip(' '))
+
+    return (
+        leading_spaces == required_spacing
+        and trailing_spaces == required_spacing)
+
+
+class Jinja2Bear(LocalBear):
+    LANGUAGES = {"Jinja2"}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    ASCIINEMA_URL = 'https://asciinema.org/a/7i5xxgwf20hkb3gz8jvdxe1hu'
```

Comment:
nice, I'd use a more common aspect ratio (less height) so it can be properly viewed on all devices. This is super cool.